docs/explanation/upgrade.rst

```diff
@@ -3,6 +3,30 @@ Upgrade
 =======
 
 This phase is responsible for applying the upgrade plan step by step (see
-:doc:`Upgrade a cloud <../how-to/upgrade-cloud>`). Most steps will be applied
-sequentially, but for some that are defined to be run in parallel, they will be
-executed in such a way.
+:doc:`Upgrade a cloud <../how-to/upgrade-cloud>`).
+
+For the **control-plane**, COU upgrades applications in a strict sequential
+manner, following a predefined order (see :doc:`Upgrade groups <./upgrade-groups>`) and
+halting the process should any application encounter an upgrade failure. All
```

Comment:
```suggestion
halting the process if any application encounter an upgrade failure. All
```

Reply:
These two sentences actually have the same meaning (`should` is a bit more formal). I'm keeping it this way for now if you don't have a strong opinion for change it.